hadoop-hdds/common/src/main/java/org/apache/hadoop/ozone/common/ChecksumByteBufferImpl.java

```diff
@@ -35,6 +39,8 @@ public class ChecksumByteBufferImpl implements ChecksumByteBuffer {
   private final Checksum checksum;
```

Comment:
I don't understand. `ChecksumByteBufferImpl` is a decoration of an actual `Checksum` to simulate (and now bridge) the `update(ByteBuffer)`.

Reply:
Sorry, my previous comment does not make sense.  My suggestion is not to use `CRC32` and `PureJavaCrc32C`.

https://github.com/apache/ozone/blob/5dd14acf4f0826fdb795866c274db301a7682981/hadoop-hdds/common/src/main/java/org/apache/hadoop/ozone/common/ChecksumByteBufferFactory.java#L74-L76
We may
```java
return new PureJavaCrc32ByteBuffer();
```

Similarly, 
https://github.com/apache/ozone/blob/5dd14acf4f0826fdb795866c274db301a7682981/hadoop-hdds/common/src/main/java/org/apache/hadoop/ozone/common/ChecksumByteBufferFactory.java#L88
We may
```java
return new PureJavaCrc32CByteBuffer();
```

Reply:
`PureJavaCrc32ByteBuffer` impl involves reading buffers, while the  `CRC32` implementation in JDK9+ (e.g. in OpenJdk11) calculates checksum using native code on the native buffer directly and that probably indicates performance advantages. 

Probably we should only replace the `return new ChecksumByteBufferImpl(new PureJavaCrc32C()); ` by  `return new PureJavaCrc32CByteBuffer();`, but I'm not sure what's the advatages.

Reply:
> ...  implementation in JDK9+ (e.g. in OpenJdk11) calculates using native code on the native buffer directly and that probably indicates performance advantages.

It is likely.  We should benchmark it.

> ... but I'm not sure what's the advatages.

The advantage is to avoid creating an array and copying for (slow path) direct buffers in Java 8.

Reply:
+1 I'll merge it.